packages/atomic/cypress/integration/segmented-facet-scrollable-actions.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  TagProps,
+  TestFixture,
+  generateComponentHTML,
+  addElement,
+} from '../fixtures/test-fixture';
+import {segmentedFacetComponent} from './facets/segmented-facet/segmented-facet-selectors';
+import {
+  scrollableComponent,
+  ScrollableSelectors,
+} from './segmented-facet-scrollable-selectors';
+
+type direction = 'right' | 'left';
+
+let scrollable!: HTMLElement;
+
+function appendSegmentedFacet(facetProps: TagProps) {
+  scrollable.append(
+    generateComponentHTML('atomic-segmented-facet', facetProps)
+  );
+}
+
+export const addScrollable =
+  (facetProps: TagProps = {}) =>
+  (env: TestFixture) => {
+    scrollable = generateComponentHTML('atomic-segmented-facet-scrollable');
+    appendSegmentedFacet(facetProps);
+    addElement(env, scrollable);
+  };
```

Comment:
Instead of appending segmented facets, you could take children as a prop (similar to `addResultList`) and `appendSegmentedFacet` could instead return the generated element (similar to `buildTemplateWithoutSections`). An example would be `result-date.cypress.ts`'s `addResultDateInResultList`.

Reply:
I think that `addResultDateInResultList `'s code is more convoluted and harder to follow.